### PR TITLE
Sync agent json patches

### PR DESCRIPTION
**Pull request by philbrookes**
Primary changes:
- Implements specdiff annotations
- Removes MCTC annotations from downstream resources

## verification
- MCTC_WORKLOAD_CLUSTERS_COUNT=1 make local-setup
- export KUBECONFIG=./tmp/kubeconfigs/mctc-workload-1.kubeconfig
- make kind-load-syncer
- make deploy-syncer
- export KUBECONFIG=./tmp/kubeconfigs/mctc-control-plane.kubeconfig
- kubectl create namespace mctc-tenant
- kubectl apply -n mctc-tenant -f ./config/samples/gateway.yaml
- export KUBECONFIG=./tmp/kubeconfigs/mctc-workload-1.kubeconfig
- kubectl get gateways example-gateway -n mctc-downstream -o yaml
- see that the name of the listener is 'test' and the gatewayClassName is 'boo'
- see that no mctc specific annotations exist

### CODE REVIEW

**Comment by sergioifg94** [comment]
@philbrookes Verified it all works as expected. Only thing I noticed is that it took 4 minutes for the gateway to be synced downstream after creating it in the control cluster. Not sure if this is expected

**Comment by david-martin** [comment]
@philbrookes I do intend to review the changes here.
I'm focusing on getting #110 landed first so I can tag milestone-2 with the walkthrough working. If you don't mind, I'd like to hold off on merging this until at least #110 is merged, just in case there's any knock on with the walkthrough steps.
Thanks

**Comment by philbrookes** [comment]
> @philbrookes I do intend to review the changes here. I'm focusing on getting #110 landed first so I can tag milestone-2 with the walkthrough working. If you don't mind, I'd like to hold off on merging this until at least #110 is merged, just in case there's any knock on with the walkthrough steps. Thanks

yes, I had thought that was how we'd progress these PRs. Not sure if we can use the merge queue to reflect that somehow?

**Comment by philbrookes** [comment]
> @philbrookes Verified it all works as expected. Only thing I noticed is that it took 4 minutes for the gateway to be synced downstream after creating it in the control cluster. Not sure if this is expected

hmm, that sounds a little strange, it was much quicker for me, when I ran it locally. I'll try and recreate that

**Comment by david-martin** [comment]
👀

**Review by david-martin — APPROVED** [review]
Verified locally.
The listener name gets transformed OK.
However, the gatewayClassName gets set to `istio`. This is expected right now as there's a hardcoded transform that can move to an annotation transform after this merges.

Let's get this merged.

**Pull request merged**
